compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -864,6 +863,42 @@ class ConvertHALInterfaceBindingSubspanOp : public ConvertToLLVMPattern {
   }
 };
 
+/// Convert memref.extract_aligned_pointer_as_index to the pointer recovered
+/// from a HAL::InterfaceBindingSubspanOp.
+class ConvertExtractAlignedPointerAsIndex : public ConvertToLLVMPattern {
+ public:
+  explicit ConvertExtractAlignedPointerAsIndex(MLIRContext *context,
+                                               LLVMTypeConverter &converter)
+      : ConvertToLLVMPattern(
+            memref::ExtractAlignedPointerAsIndexOp::getOperationName(), context,
+            converter) {}
+
+  LogicalResult matchAndRewrite(
+      Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto llvmFuncOp = op->getParentOfType<LLVM::LLVMFuncOp>();
+    if (!llvmFuncOp) {
+      return rewriter.notifyMatchFailure(op, "no parent LLVM func");
+    }
+
+    auto extractOp = cast<memref::ExtractAlignedPointerAsIndexOp>(op);
+    auto bindingOp = extractOp.getSource()
+                         .getDefiningOp<IREE::HAL::InterfaceBindingSubspanOp>();
+    if (!bindingOp) {
+      return rewriter.notifyMatchFailure(op, [&](Diagnostic &diag) {
+        diag << "not an InterfaceBindingSubspanOp source: "
+             << extractOp.getSource();
+      });
+    }
+    HALDispatchABI abi(llvmFuncOp, getTypeConverter());
+    Value baseMemrefPtr = abi.loadBindingPtr(
```

Comment:
you'll need to offset this by bindingOp.getByteOffset() (the pointer here is just the base)